pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/EventService.java

```diff
@@ -0,0 +1,177 @@
+        return triggerDssWithEvent(kitDDPNotification.getEventName(), ddpInstance,
+                kitDDPNotification.getDate() / 1000, kitDDPNotification.getParticipantId(),
+                kitDDPNotification.getDdpKitRequestId(), kitDDPNotification.getUploadReason());
+    }
+
+    /**
+     * Tries to send a POST request to DSS to trigger the event for a kit or a participant,
+     * It will try MAX_TRIES times before giving up and adding the event to the EVENT_QUEUE table.
+     * @param eventType - the type of event to be triggered, e.g. "SALIVA_RECEIVED"
+     * @param eventInfo - the information to be sent in the event, for kit events it is the "ddp_kit_request_id",
+     *      and for participant events it is the ddp_participant_id
+     * @return true if the request was successful, false otherwise
+     * */
+    @VisibleForTesting
+    protected static boolean triggerDssWithEvent(@NonNull String eventType, DDPInstance ddpInstance, long eventDate,
+                                        @NotNull String ddpParticipantId, @NotNull String eventInfo, String reason) {
+        for (int tries = 0; tries < MAX_TRIES; tries++) {
+            try {
+                if (sendDDPEventRequest(eventType, ddpInstance, eventDate, ddpParticipantId, eventInfo, reason)) {
+                    return true;
+                }
+            } catch (Exception e) {
+                logTriggerFailure(ddpInstance, eventType, ddpParticipantId, eventInfo, tries, e);
+            }
+        }
+        logTriggerExhausted(ddpInstance, eventType, ddpParticipantId, eventInfo);
+        return false;
+    }
+
+    /**
+     * Sends the POST request to DSS to trigger the event for a kit or a participant.
+     * */
+    @VisibleForTesting
+    protected static boolean sendDDPEventRequest(String eventType, DDPInstance ddpInstance, long eventDate, String ddpParticipantId,
+                                        String eventInfo, String reason) throws Exception {
+        Event event = new Event(ddpParticipantId, eventType, eventDate, reason, eventInfo);
+        String sendRequest = ddpInstance.getBaseUrl() + RoutePath.DDP_PARTICIPANT_EVENT_PATH + "/" + ddpParticipantId;
+        int responseCode = DDPRequestUtil.postRequest(sendRequest, event, ddpInstance.getName(), ddpInstance.isHasAuth0Token());
+        if (responseCode == 200) {
+            return true;
+        }
+        logger.error("Bad Response: Failed to trigger {} to notify participant {} about {} for dsm_kit_request_id {}, response code {}",
+                ddpInstance.getName(), ddpParticipantId, eventType, eventInfo, responseCode);
+        return false;
+    }
+
+    /**
+     * logs the failure of the trigger event to DSS
+     * */
+    @VisibleForTesting
+    protected static void logTriggerFailure(DDPInstance ddpInstance, String eventType, String ddpParticipantId, String eventInfo, int tries,
```

Comment:
This method is called if an exception is thrown while sending a post request to DSS, which will be logged in here